File: kafka/src/main/java/com/example/kafka/controller/KafkaController.java
package com.example.kafka.controller;

import com.example.kafka.service.KafkaConsumerService;
import com.example.kafka.service.KafkaProducerService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class KafkaController {
  private final KafkaProducerService kafkaProducerService;
  private final KafkaConsumerService kafkaConsumerService;
  public KafkaController(KafkaProducerService kafkaProducerService,
      KafkaConsumerService kafkaConsumerService) {
    this.kafkaProducerService = kafkaProducerService;
    this.kafkaConsumerService = kafkaConsumerService;
  }

  @GetMapping("/send")
  public String sendMessage(@RequestParam String topic, @RequestParam String message) {
    kafkaProducerService.sendMessage(topic, message);
    return  "Message sent to topic: " + topic;
  }

}

File: kafka/src/main/kotlin/com/example/kafka/controller/KafkaController.kt
package com.example.kafka.controller

import com.example.kafka.dto.ProductEvent
import com.example.kafka.service.KafkaProducerService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/kafka")
class KafkaController(
    private val kafkaProducerService: KafkaProducerService
) {

    @PostMapping("/send")
    fun sendMessage(event: ProductEvent): String {
        kafkaProducerService.sendMessage(event)
        return "send Message Complete"
    }
}